cluster-operator/src/main/java/io/strimzi/operator/cluster/model/dynamicconfig/KafkaBrokerConfigurationBuilder.java

```diff
@@ -0,0 +1,360 @@
+                options.add("oauth.client.secret=\"${STRIMZI_" + listenerNameInEnvVar + "_OAUTH_CLIENT_SECRET}\"");
+            }
+
+            if (oauth.getTlsTrustedCertificates() != null && oauth.getTlsTrustedCertificates().size() > 0)    {
+                options.add(String.format("oauth.ssl.truststore.location=\"/tmp/kafka/oauth-%s.truststore.p12\"", listenerNameInProperty));
+                options.add("oauth.ssl.truststore.password=\"${CERTS_STORE_PASSWORD}\"");
+                options.add("oauth.ssl.truststore.type=\"PKCS12\"");
+            }
+
+            writer.println(String.format("listener.name.%s.oauthbearer.sasl.server.callback.handler.class=io.strimzi.kafka.oauth.server.JaasServerOauthValidatorCallbackHandler", listenerNameInProperty));
+            writer.println(String.format("listener.name.%s.oauthbearer.sasl.jaas.config=org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required unsecuredLoginStringClaim_sub=\"thePrincipalName\" %s;", listenerNameInProperty, String.join(" ", options)));
+            writer.println(String.format("listener.name.%s.sasl.enabled.mechanisms=OAUTHBEARER", listenerNameInProperty));
+            writer.println();
+        } else if (auth instanceof KafkaListenerAuthenticationScramSha512) {
+            securityProtocol.add(String.format("%s:%s", listenerName, getSecurityProtocol(tls, true)));
+
+            writer.println(String.format("listener.name.%s.scram-sha-512.sasl.jaas.config=org.apache.kafka.common.security.scram.ScramLoginModule required;", listenerNameInProperty));
+            writer.println(String.format("listener.name.%s.sasl.enabled.mechanisms=SCRAM-SHA-512", listenerNameInProperty));
+            writer.println();
+        } else if (auth instanceof KafkaListenerAuthenticationTls) {
+            securityProtocol.add(String.format("%s:%s", listenerName, getSecurityProtocol(tls, false)));
+
+            writer.println(String.format("listener.name.%s.ssl.client.auth=required", listenerNameInProperty));
+            writer.println(String.format("listener.name.%s.ssl.truststore.location=/tmp/kafka/clients.truststore.p12", listenerNameInProperty));
+            writer.println(String.format("listener.name.%s.ssl.truststore.password=${CERTS_STORE_PASSWORD}", listenerNameInProperty));
+            writer.println(String.format("listener.name.%s.ssl.truststore.type=PKCS12", listenerNameInProperty));
+            writer.println();
+        } else {
+            securityProtocol.add(String.format("%s:%s", listenerName, getSecurityProtocol(tls, false)));
+        }
+    }
+
+    private String getSecurityProtocol(boolean tls, boolean sasl)   {
+        if (tls)    {
+            if (sasl)   {
+                return "SASL_SSL";
+            } else {
+                return "SSL";
+            }
+        } else  {
+            if (sasl)   {
+                return "SASL_PLAINTEXT";
+            } else {
+                return "PLAINTEXT";
+            }
+        }
+    }
```

Comment:
what do you think something like ...
```java
String a = tls ? "SSL" : "PLAINTEXT";
return sasl ? "SASL_" + a : a;
```
Maybe a better name for variable `a` :-)